edge/pkg/devicetwin/dmiserver/server.go

```diff
@@ -150,6 +150,26 @@ func (s *server) ReportDeviceStatus(ctx context.Context, in *pb.ReportDeviceStat
 	return &pb.ReportDeviceStatusResponse{}, nil
 }
 
+func (s *server) ReportDeviceStates(ctx context.Context, in *pb.ReportDeviceStatesRequest) (*pb.
+	ReportDeviceStatesResponse, error) {
+	if !s.limiter.Allow() {
+		return nil, fmt.Errorf("fail to report device states because of too many request: %s", in.DeviceName)
```

Comment:
The function name here is relatively close to the `ReportDeviceStates` above. You can add a comment to explain what data is reported. Another approach is to replace `ReportDeviceStates` with `ReportDeviceTwinStates` as a whole? Do you think it is appropriate?